### ISSUE TITLE
subcommands should have optional description

### ISSUE THREAD

**Issue by hsn10** [scallop/scallop#62]
demo:

 val tree = new Subcommand("tree", descr = "create banana tree") {
    val bananas = opt[Boolean]("bananas")
  }

**Comment by hsn10** [comment]
``` scala
class CommandLine(arguments: Seq[String]) extends ScallopConf(arguments) {
  version("lookaside admin tool 0.0.1")
  /** cassandra host */
  val host = opt[String](default = Some("localhost:9160"), short = 'h')
  val create = new Subcommand("create") {
    descr("create new lookaside repository")
    val name = trailArg[String]("name", descr = "lookaside repository name, must be valid cassandra keyspace name")
  }
}
```

i expect help like that:

```
lookaside admin tool 0.0.1
  -h, --host  <arg>    (default = localhost:9160)
      --help          Show help message
      --version       Show version of this program

Subcommand: create   - create new lookaside repository
      --help      Show help message

 trailing arguments:
  name (required)   lookaside repository name, must be valid cassandra keyspace
                    name
```

**Comment by Rogach** [comment]
I implemented it, and published a snapshot (for scala 2.10.x):

``` scala
resolvers += "sonatype snapshots" at "https://oss.sonatype.org/content/repositories/snapshots"
libraryDependencies += "org.rogach" %% "scallop" % "0.9.2~2~SNAPSHOT"
```
